.github/scripts/findUnusedStyles.ts

```diff
@@ -0,0 +1,462 @@
+import * as fs from 'fs';
+import {globSync} from 'glob';
+import * as path from 'path';
+import * as ts from 'typescript';
+
+type StyleDefinition = {
+    key: string;
+    file: string;
+    line: number;
+    column: number;
+};
+
+// Static patterns and constants - created once
+const STYLE_FILE_EXTENSIONS = '**/*.{ts,tsx,js,jsx}';
+const EXCLUDED_STYLE_DIRECTORIES = ['src/styles/utils/', 'src/styles/generators/', 'src/styles/theme/'];
+const VALID_IDENTIFIER_PATTERN = /^[a-zA-Z_$][a-zA-Z0-9_$]*$/;
+const SPREAD_PROPERTY_PATTERN = /\.\.\.\s*([a-zA-Z_$][a-zA-Z0-9_$]*)\s*\.\s*([a-zA-Z_$][a-zA-Z0-9_$]*)/g;
```

Comment:
I'm skeptical that you would need to use a regex for this. TS AST has entities for spread operators:

<img width="1512" height="823" alt="image" src="https://github.com/user-attachments/assets/6de325ee-3ee2-4d7e-bf3d-944ecd33b0bf" />

Reply:
I created some test files to make the sure and worked well